common/beerocks/bcl/source/beerocks_string_utils.cpp

```diff
@@ -93,8 +93,16 @@ std::vector<std::string> string_utils::str_split(const std::string &s, char deli
     return elems;
 }
 
-int64_t string_utils::stoi(std::string str, const char *calling_file, int calling_line)
+int64_t string_utils::stoi(const std::string &str, const char *calling_file, int calling_line)
 {
+    if (str.find_first_not_of("0123456789") != std::string::npos) {
```

Comment:
This will break the (currently accepted, but I don't know if it's actually used) case of `stoi("0xff")`.

I think a better solution would be to check that `val_s` reached EOF, i.e. change the condition below to `if (val_s.fail() || !val_s.eof())`.

Reply:
when we expect to convert hex value, we use std::stoi(), and explicit mention the base 16.
But I agree that maybe it will be best to refactor the use of stoi. Not in this PR of course.